Replaced moment with dayjs fixed issue #34

rohit-raje-786 opened the pull request:
Replaced moment with dayjs to fixed issue #34

rohit-raje-786 committed:
fixes : replace moment with dayjs issue #34

rohit-raje-786 committed:
fixes : replace moment with dayjs issue#34

ruddi10 commented:
@rohit-raje-786 hey actually react-datetime has a dependency on moment so when you are replacing it with dayjs isn't it causing some issues?

ruddi10 commented:
I guess currently you haven't removed moment from your dependency and that's why it's working fine but it won't work if we remove moment from our dependency

rohit-raje-786 committed:
replaced react-datetime with @material-ui/pickers issue #34

rohit-raje-786 commented:
Hey @ruddi10 have a look at it now I have replaced rect-datetime with @material-ui/pickers.And I have removed moment.js and react-datetime
<img width="291" alt="Screenshot 2022-01-22 at 12 43 49 AM" src="https://user-images.githubusercontent.com/61861303/150585784-0113e138-ac86-4dc9-a2ff-648671bd8100.png">

ruddi10 commented:
@aquibbaig I think in the older version of the dashboard we were using material-ui/pickers and there were some reasons that the maintainers wanted to go with react-datetime so what's your take on using something other than this package ? If so do you have some good alternative in your mind?

aquibbaig commented on .gitpod.yml:
```diff
@@ -0,0 +1,11 @@
+# This configuration file was automatically generated by Gitpod.
```
shouldn't be a part of this PR

aquibbaig commented on src/components/TimeQuerier/index.tsx:
```diff
@@ -13,28 +13,35 @@ import {
   const { defaultStepValue, minStepValue, dateFormat, timeFormat } = constants;
-  const [startTime, setStartTime] = useState(moment(selectedStartTimestamp));
-  const [endTime, setEndTime] = useState(moment(selectedEndTimestamp));
+  const [startTime, setStartTime] = useState(dayjs(selectedStartTimestamp));
+  const [endTime, setEndTime] = useState(dayjs(selectedEndTimestamp));
```
I'd say keep the start and end times in the redux store (and persist them too) so that it is maintained across refreshes (mostly people do not like dates getting reset)

aquibbaig commented on src/components/TimeQuerier/index.tsx:
```diff
@@ -44,13 +51,13 @@ const TimeQuerier: React.FC = () => {
   };
-  const valid = (current: moment.Moment) => {
-    return current.isBefore(moment());
+  const valid = (current: dayjs.Dayjs) => {
```
use the `utils/` folder

aquibbaig commented on src/components/TimeQuerier/index.tsx:
```diff
@@ -13,28 +13,35 @@ import {
+  const handleEndChange = (date: dayjs.Dayjs | string) => {
     if (typeof date !== "string") {
       setEndTime(date);
       setError(undefined);
```
error is either `null` or `Error`

aquibbaig reviewed (changes requested):
Hey, thanks for this PR, I am posting some initial feedback on your work

<img width="282" alt="Screenshot 2022-02-06 at 2 43 22 AM" src="https://user-images.githubusercontent.com/26324376/152659294-1aeff312-181c-408a-95f9-da51b02a4b16.png">

1. Start Time -> Picker should be in a center-aligned vertically
2. For the consistent look and feel, use the same font family and same font colors in pickers.
3. Remove the background on click of the picker (or come up w a better design)
